include/keisan/number.hpp

```diff
@@ -61,6 +61,9 @@ T wrap(const T & value, const T & min, const T & max);
 template<typename T, enable_if_is_integral<T> = true>
 T wrap(const T & value, const T & min, const T & max);
 
+template<typename T, enable_if_is_floating_point<T> = true>
+T smooth(T value, T target, T ratio);
```

Comment:
just use `template<typename T>` here

actually `template<typename T, enable_if_is_floating_point<T> = true>` is a tricky approach to enable different implementation of a function when using different type (like wrap, it has different approach when using integers and floats)

TBH, i actually hate that approach 😆

Reply:
done